monitor-manager/monitor-manager.py

```diff
@@ -0,0 +1,76 @@
+from csv import reader
+from flask import Flask
+import socket
+import json
+import requests
+import time
+
+app = Flask(__name__)
+
+
+@app.route('/', methods=['get'])
+def health_check():
+    return "Monitor Manager working"
+
+
+@app.route('/experiments', methods=['post'])
+def start_experiments():
+    experiments_plan = read_csv("experiments-plan.csv")
+    hosts_info = read_csv("hosts-info.csv")
+    is_spark_node, host_ip = check_if_spark_node(hosts_info)
+
+    for function_name, table in experiments_plan:
+        print("Task submitting")
+        submit_response = requests.post("http://192.168.55.20:5000/submit", json={"function_name": function_name})
+        submit_response = json.loads(submit_response.text)
+        driver_id = submit_response["submissionId"]
+        print("Submit driver ID: ", driver_id)
+
+        print("Start monitoring")
+        system_monitor(host_ip, hosts_info, function_name, True)
+        #
+        status = "RUNNING"
+        while status == "RUNNING":

```

Comment:
I don't understand why this loop isn't inside `system_monitor` function? Boolean parameters aren't good practice.

Also, the way you implemented it right now `system_monitor` behavior varies depending on the boolean parameter - ani-pattern.

IMHO you could encapsulate whole logic in `system_monitor` function which will be as follows
1. Begin "for" loop
2. prepare `url` and `request_data`
3. Start monitoring
4. Escape the for loop
5. While loop with getting the driver status.
6. For loop in which you stop monitoring at all nodes

Advantages of proposed change:
- no if else statement
- no boolean param 
- system monitor function keeps all responsibility of the process

Reply:
Good idea, i will change that